protocol/message_persistence.go

```diff
@@ -2420,15 +2419,16 @@ func getMessagesAndCursorsFromScanRows(db sqlitePersistence, rows *sql.Rows) ([]
 		if msg, ok := messageIdx[message.ID]; !ok {
 			messageIdx[message.ID] = &message
 			cursors = append(cursors, cursor)
+			messageIDs = append(messageIDs, message.ID)
```

Comment:
Could we append `message` directly to `var messages common.Messages`? That way the conversion loop is no longer required

Reply:
Something like below:

```go
	var cursors []string
	var messages common.Messages

	messageIdx := make(map[string]*common.Message, 0)

	for rows.Next() {
		// There's a possibility of multiple rows per message if the
		// message has a discordMessage and the discordMessage has multiple
		// attachments
		//
		// Hence, we make sure we're aggregating all attachments on a single
		// common.Message
		var (
			message common.Message
			cursor  string
		)
		if err := db.tableUserMessagesScanAllFields(rows, &message, &cursor); err != nil {
			return nil, nil, err
		}
		if msg, ok := messageIdx[message.ID]; !ok {
			messageIdx[message.ID] = &message
			cursors = append(cursors, cursor)
			message = append(messages, message)
```

Reply:
I think we can if the update to `messageIdx[message.ID]` in the `else if` case will also update the same item in `messages` (via reference)